### PR TITLE
mon/OSDMonitor: Add trivial sanity checks in prepare_new_pool

### PR DESCRIPTION
**Pull request by shinobu-x**
Signed-off-by: Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

**Comment by shinobu-x** [comment]
test this please

File: src/mon/OSDMonitor.cc [hunk 1]
```diff
@@ -5480,13 +5480,13 @@ int OSDMonitor::prepare_new_pool(string& name, uint64_t auid,
     pg_num = g_conf->get_val<uint64_t>("osd_pool_default_pg_num");
   if (pgp_num == 0)
     pgp_num = g_conf->get_val<uint64_t>("osd_pool_default_pgp_num");
-  if (pg_num > (unsigned)g_conf->mon_max_pool_pg_num) {
+  if (pg_num == 0 || pg_num > (unsigned)g_conf->mon_max_pool_pg_num) {
     *ss << "'pg_num' must be greater than 0 and less than or equal to "
         << g_conf->mon_max_pool_pg_num
-        << " (you may adjust 'mon max pool pg num' for higher values)";
+        << " (you may adjust 'mon_max_pool_pg_num' for higher values)";
```
Comment (joscollin) [inline src/mon/OSDMonitor.cc]: Unnecessary change

File: src/mon/OSDMonitor.cc [hunk 2]
```diff
@@ -5480,13 +5480,13 @@ int OSDMonitor::prepare_new_pool(string& name, uint64_t auid,
...
     return -ERANGE;
   }
-  if (pgp_num > pg_num) {
+  if (pgp_num == 0 || pgp_num > pg_num) {
```
Comment (joscollin) [inline src/mon/OSDMonitor.cc]: I think this is the correct way.
`if (! (pgp_num > 0 && pgp_num <= pg_num))`

File: src/mon/OSDMonitor.cc [hunk 3]
```diff
@@ -5480,13 +5480,13 @@ int OSDMonitor::prepare_new_pool(string& name, uint64_t auid,
     pg_num = g_conf->get_val<uint64_t>("osd_pool_default_pg_num");
   if (pgp_num == 0)
     pgp_num = g_conf->get_val<uint64_t>("osd_pool_default_pgp_num");
-  if (pg_num > (unsigned)g_conf->mon_max_pool_pg_num) {
+  if (pg_num == 0 || pg_num > (unsigned)g_conf->mon_max_pool_pg_num) {
```
Comment (joscollin) [inline src/mon/OSDMonitor.cc]: same as the comment on L5489

File: src/mon/OSDMonitor.cc [hunk 4]
```diff
@@ -5480,13 +5480,13 @@ int OSDMonitor::prepare_new_pool(string& name, uint64_t auid,
...
     *ss << "'pg_num' must be greater than 0 and less than or equal to "
         << g_conf->mon_max_pool_pg_num
-        << " (you may adjust 'mon max pool pg num' for higher values)";
+        << " (you may adjust 'mon_max_pool_pg_num' for higher values)";
     return -ERANGE;
   }
-  if (pgp_num > pg_num) {
+  if (pgp_num == 0 || pgp_num > pg_num) {
     *ss << "'pgp_num' must be greater than 0 and lower or equal than 'pg_num'"
         << ", which in this case is " << pg_num;
```
Comment (joscollin) [inline src/mon/OSDMonitor.cc]: This text is not proper. Change as per https://github.com/shinobu-x/ceph/blob/38215d01bb9f01d367bf2da70ade6c1d7539a884/src/mon/OSDMonitor.cc#L5484.

`greater than 0 and less than or equal to`

**Comment by gregsfortytwo** [comment]
Closing this since it has outstanding review comments and conflicts.
